api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPersonServiceImpl.java

```diff
@@ -68,4 +72,13 @@ public Person getPersonByUuid(String uuid) {
 		return fhirPersonDao.findPersonsByGender(gender).stream().map(personTranslator::toFhirResource)
 		        .collect(Collectors.toList());
 	}
+	
+	@Override
+	public Collection<Person> searchForPeople(StringOrListParam name, TokenOrListParam identifier, TokenOrListParam gender,
+	        DateRangeParam birthDate, StringOrListParam city, StringOrListParam state, StringOrListParam postalCode,
+	        SortSpec sort) {
+		// TODO Auto-generated method stub
```

Comment:
@CaptainDaVinci I think we'll have to write something like what's written in `FhirObservationDaoImpl` @ibacher Please guide

```java
@Override
	public Collection<Obs> searchForObservations(ReferenceParam encounterReference, ReferenceParam patientReference,
	        TokenAndListParam code, SortSpec sort) {
		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Obs.class);
		
		handleEncounterReference("e", encounterReference).ifPresent(c -> criteria.createAlias("encounter", "e").add(c));
		handlePatientReference(criteria, patientReference);
		handleCodedConcept(criteria, code);
		handleSort(criteria, sort);
		
		return criteria.list();
	}
```

Reply:
Yes, I'm working on it. I'm using PatientDaoImpl as a reference.